src/context/engagement_context.tsx

```diff
@@ -97,44 +107,40 @@ export const EngagementProvider = ({
 
   const saveEngagement = useCallback(
     async (data: any) => {
+      feedbackContext.showLoader();
       const oldEngagement = _updateEngagementInPlace(data);
       try {
         const returnedEngagement = await engagementRepository.saveEngagement(
           data
         );
+        feedbackContext.hideLoader();
         _updateEngagementInPlace(returnedEngagement);
       } catch (e) {
         _updateEngagementInPlace(oldEngagement);
-        // TODO: Add error state
+        feedbackContext.hideLoader();
+        feedbackContext.showAlert("There was an issue with saving your changes. Please followup with an administrator if this continues.", "error");
       }
     },
-    [engagementRepository, _updateEngagementInPlace]
+    [engagementRepository, _updateEngagementInPlace, feedbackContext]
   );
 
-  const showSuccessMessage = () => {
-    console.log("success");
-    //TODO: Once interaction feedback system is worked out - implement here
-  }
-
-  const showErrorMessage = () => {
-    console.log("error");
-    //TODO: Once interaction feedback system is worked out - implement here
-  }
-
   const launchEngagement = useCallback(
     async (data: any) => {
+      feedbackContext.showLoader();
       const oldEngagement = _updateEngagementInPlace(data);
       try {
         const returnedEngagement = await engagementRepository.launchEngagement(data);
         _updateEngagementInPlace(returnedEngagement);
-        showSuccessMessage();
         setActiveEngagement(returnedEngagement);
+        feedbackContext.hideLoader();
+        feedbackContext.showAlert("You have successfully launched your cluster!", "success");
```

Comment:
Might want to have a conversation in the parking lot about use of the term "cluster" throughout the app in general, but otherwise (technically) looks great! Think we're holding off on merging new features but let's also ask about that.

Reply:
I think that this is the last being merged... Will clarify in standup. Thanks Jacob